### PR TITLE
fix(error): attach command response to MongoWriteConcernError

### PR DESCRIPTION
**Pull request by kvwalker**
Fixes NODE-1521

### CODE REVIEW

File: lib/error.js [hunk 1]
```diff
@@ -131,12 +131,17 @@ function isRetryableError(error) {
...
  * @param {Error|string|object} message The error message
  * @property {string} message The error message
+ * @property {object} result The result document (provided if ok: 1)
  * @return {MongoWriteConcernError} A MongoWriteConcernError instance
  * @extends {MongoError}
  */
-const MongoWriteConcernError = function(message) {
-  MongoError.call(this, message);
+const MongoWriteConcernError = function(message, result) {
+  MongoError.call(this, message, result);
```
Comment (daprahamian) [inline lib/error.js]: Why do we need to pass `result` to `MongoError`? It doesn't appear to do anything with it.

File: lib/connection/pool.js [hunk 2]
```diff
@@ -581,6 +581,13 @@ function messageHandler(self) {
           }
 
           if (responseDoc.writeConcernError) {
+            if (responseDoc.ok === 1) {
```
Comment (daprahamian) [inline lib/connection/pool.js]: What about doing a ternary to create the error, and then using a single callback?

```js
if (responseDoc.writeConcernError) {
  const err = responseDoc.ok === 1 ?
    new MongoWriteConcernError(responseDoc.writeConcernError, responseDoc) :
    new MongoWriteConcernError(responseDoc.writeConcernError);
  return handleOperationCallback(self, workItem.cb, err);
}
```

File: lib/error.js (hunk 1 above)
Comment (kvwalker) [reply lib/error.js]: oops, thanks for catching that!

File: lib/error.js [hunk 3]
```diff
@@ -131,12 +131,17 @@ function isRetryableError(error) {
  * @class
  * @param {Error|string|object} message The error message
  * @property {string} message The error message
+ * @property {object} result The result document (provided if ok: 1)
```
Comment (mbroadst) [inline lib/error.js]: Please wrap the name of the argument with brackets to indicate it is optional, and add it to the `@param` documentation.

**Pull request merged**